climada_petals/entity/impact_funcs/river_flood.py

```diff
@@ -55,154 +53,452 @@ def __init__(self):
-                       "Use ImpfRiverFlood.from_region instead.")
-        self.__dict__ = ImpfRiverFlood.from_region("Asia", *args, **kwargs).__dict__
+                       "Use ImpfRiverFlood.from_region_sector instead.")
+        self.__dict__ = ImpfRiverFlood.from_jrc_region_sector("Asia", *args, **kwargs).__dict__
 
     def set_RF_Impf_Europe(self, *args, **kwargs):
-        """This function is deprecated, use ImpfRiverFlood.from_region instead."""
+        """This function is deprecated, use ImpfRiverFlood.from_region_sector instead."""
         LOGGER.warning("The use of ImpfRiverFlood.set_RF_Impf_Europe is deprecated."
-                       "Use ImpfRiverFlood.from_region instead.")
-        self.__dict__ = ImpfRiverFlood.from_region("Europe", *args, **kwargs).__dict__
+                       "Use ImpfRiverFlood.from_region_sector instead.")
+        self.__dict__ = ImpfRiverFlood.from_jrc_region_sector("Europe", *args, **kwargs).__dict__
 
     def set_RF_Impf_NorthAmerica(self, *args, **kwargs):
-        """This function is deprecated, use ImpfRiverFlood.from_region instead."""
+        """This function is deprecated, use ImpfRiverFlood.from_region_sector instead."""
         LOGGER.warning("The use of ImpfRiverFlood.set_RF_Impf_NorthAmerica is deprecated."
-                       "Use ImpfRiverFlood.from_region instead.")
-        self.__dict__ = ImpfRiverFlood.from_region("NorthAmerica", *args, **kwargs).__dict__
+                       "Use ImpfRiverFlood.from_region_sector instead.")
+        self.__dict__ = ImpfRiverFlood.from_jrc_region_sector("NorthAmerica", *args, **kwargs).__dict__
 
     def set_RF_Impf_Oceania(self, *args, **kwargs):
-        """This function is deprecated, use ImpfRiverFlood.from_region instead."""
+        """This function is deprecated, use ImpfRiverFlood.from_region_sector instead."""
         LOGGER.warning("The use of ImpfRiverFlood.set_RF_Impf_Oceania is deprecated."
-                       "Use ImpfRiverFlood.from_region instead.")
-        self.__dict__ = ImpfRiverFlood.from_region("Oceania", *args, **kwargs).__dict__
+                       "Use ImpfRiverFlood.from_region_sector instead.")
+        self.__dict__ = ImpfRiverFlood.from_jrc_region_sector("Oceania", *args, **kwargs).__dict__
 
     def set_RF_Impf_SouthAmerica(self, *args, **kwargs):
-        """This function is deprecated, use ImpfRiverFlood.from_region instead."""
+        """This function is deprecated, use ImpfRiverFlood.from_region_sector instead."""
         LOGGER.warning("The use of ImpfRiverFlood.set_RF_Impf_SouthAmerica is deprecated."
-                       "Use ImpfRiverFlood.from_region instead.")
-        self.__dict__ = ImpfRiverFlood.from_region("SouthAmerica", *args, **kwargs).__dict__
+                       "Use ImpfRiverFlood.from_region_sector instead.")
+        self.__dict__ = ImpfRiverFlood.from_jrc_region_sector("SouthAmerica", *args, **kwargs).__dict__
+
+def from_jrc_impf_residential(region):
```

Comment:
This would probably have been much better solved with a dataclass. However, I am not sure it is worth the effort now. @emanuel-schmid what do you think?